sydent/db/hashing_metadata.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class HashingMetadataStore:
+    def __init__(self, sydent):
+        self.sydent = sydent
+
+    def get_lookup_pepper(self):
+        """Return the value of the current lookup pepper from the db
+        
+        :returns a pepper if it exists in the database, or None if one does
+                 not exist
+        """
+        cur = self.sydent.db.cursor()
+        res = cur.execute("select lookup_pepper from hashing_metadata")
+        row = res.fetchone()
+
+        if not row:
+            return None
+        return row[0]
+
+    def store_lookup_pepper(self, hashing_function, pepper):
+        """Stores a new lookup pepper in the hashing_metadata db table and rehashes all 3PIDs
+
+        :param hashing_function: A function with single input and output strings
+        :type hashing_function func(str) -> str
+
+        :param pepper: The pepper to store in the database
+        :type pepper: str
+        """
+        cur = self.sydent.db.cursor()
+
+        # Create or update lookup_pepper
+        sql = (
+            'INSERT OR REPLACE INTO hashing_metadata (id, lookup_pepper) '
```

Comment:
You have an id column here but the table definition doesn't have an id

Reply:
https://github.com/matrix-org/sydent/pull/184/files#diff-5597f624bef7933a2e27666bb59c275eR22 ?

Reply:
oh, it's a mismatch between the migration and initial schema - have commented there